src/llir/abi.rs

```diff
@@ -124,6 +123,21 @@ impl ArgEncoding {
 
         Impl(self)
     }
+
+    pub fn contributes_to_param_mask(&self) -> bool {
+        !matches!(self, Self::Padding)
+    }
+    
+    pub fn is_immediate(&self) -> bool {
+        matches!(self,
+            | Self::String { .. }
+            | Self::JumpOffset
+            | Self::JumpTime
+            | Self::Padding
+            | Self::Integer { immediate: true, .. }
+            | Self::Float { immediate: true, .. }
+        )
```

Comment:
I think I want this one to be a full `match`.  It can be formatted like the one for getting the type below